google/cloud/internal/filesystem.h

```diff
@@ -167,6 +167,9 @@ inline bool is_other(file_status s) noexcept {
 std::uintmax_t file_size(std::string const& path);
 std::uintmax_t file_size(std::string const& path, std::error_code& ec) noexcept;
 
+/// Append @p path (which may have subdirectories) to @p directory
```

Comment:
Whether `path` can be absolute seems more important than whether it can include directories.

Reply:
Fixed